py/server/deephaven/filters.py

```diff
@@ -88,21 +97,163 @@ def not_(filter_: Filter) -> Filter:
+    the US-ASCII charset are being matched.
+
+    Unicode-aware case folding can also be enabled via the embedded flag expression (?u).
+
+    Specifying this flag may impose a performance penalty.
+    """
+
+    CANON_EQ = _JPattern.CANON_EQ
+    """
+    Enables canonical equivalence.
+
+    When this flag is specified then two characters will be considered to match if, and only if, their full canonical
+    decompositions match. The expression "a\u030A", for example, will match the string "\u00E5" when this flag is
+    specified. By default, matching does not take canonical equivalence into account.
+
+    There is no embedded flag character for enabling canonical equivalence.
+
+    Specifying this flag may impose a performance penalty.
+    """
+
+    UNIX_LINES = _JPattern.UNIX_LINES
+    """
+    Enables Unix lines mode.
+
+    In this mode, only the '\n' line terminator is recognized in the behavior of ., ^, and $.
+
+    Unix lines mode can also be enabled via the embedded flag expression (?d).
+    """
+
+    LITERAL = _JPattern.LITERAL
+    """
+    Enables literal parsing of the pattern.
+
+    When this flag is specified then the input string that specifies the pattern is treated as a sequence of literal
+    characters. Metacharacters or escape sequences in the input sequence will be given no special meaning.
+
+    The flags CASE_INSENSITIVE and UNICODE_CASE retain their impact on matching when used in conjunction with this flag.
+    The other flags become superfluous.
+
+    There is no embedded flag character for enabling literal parsing.
+    """
+
+    UNICODE_CHARACTER_CLASS = _JPattern.UNICODE_CHARACTER_CLASS
+    """
+    Enables the Unicode version of Predefined character classes and POSIX character classes.
+
+    When this flag is specified then the (US-ASCII only) Predefined character classes and POSIX character classes are in
+    conformance with Unicode Technical Standard #18: Unicode Regular Expression Annex C: Compatibility Properties.
+
+    The UNICODE_CHARACTER_CLASS mode can also be enabled via the embedded flag expression (?U).
+
+    The flag implies UNICODE_CASE, that is, it enables Unicode-aware case folding.
+
+    Specifying this flag may impose a performance penalty.
+    """
+
+    COMMENTS = _JPattern.COMMENTS
+    """
+    Permits whitespace and comments in pattern.
+
+    In this mode, whitespace is ignored, and embedded comments starting with # are ignored until the end of a line.
+
+    Comments mode can also be enabled via the embedded flag expression (?x).
+    """
+
+    @staticmethod
+    def bitwise_or(flags: Union[PatternFlag, List[PatternFlag]]) -> int:
+        return functools.reduce(
+            lambda x, y: x | y, [flag.value for flag in to_sequence(flags)]
+        )
+
+    def __repr__(self):
+        return self.name
+
+
+def pattern(
+    mode: PatternMode,
+    col: str,
+    regex: str,
+    flags: Union[PatternFlag, List[PatternFlag]] = [],
```

Comment:
```suggestion
    flags: Union[PatternFlag, List[PatternFlag]] = None,
```
Though not a big deal here, it is generally considered a safer practice to default mutable parameters to None